autoload/go/test.vim

```diff
@@ -6,7 +6,10 @@ set cpo&vim
 " compile the tests instead of running them (useful to catch errors in the
 " test files). Any other argument is appended to the final `go test` command.
 function! go#test#Test(bang, compile, ...) abort
-  let args = ["test", '-tags', go#config#BuildTags()]
+  let args = ["test"]
+  if go#config#BuildTags() > 0
+    call extend(args, ["-tags", go#util#Shellescape(tags)])
+  endif
```

Comment:
This shouldn't change. Please revert this; if you'll make the change to pass a list instead of a string as the terminal command, I think you'll find that this change is unnecessary.

Reply:
@bhcleek this look okay now ?

Reply:
It's close. This does not nee to use`go#util#Shelljoin` now, because there's no need to shell escape it since it won't be processed by a shell. `go#config#BuildTags()` returns the space separated list of tags, so using its return value unaltered should be fine.

```
 let args = ["test"]
  if go#config#BuildTags() > 0
    call extend(args, ["-tags", go#config#BuildTags()])
  endif
```